File: Personal/ComicAndPicture/ComicAndPicture/Views/ComicCell.m
//
//  ComicCell.m
//  ComicAndPicture
//
//  Created by MS on 16/3/4.
//  Copyright © 2016年 JFY. All rights reserved.
//

#import "ComicCell.h"
#import <UIImageView+WebCache.h>
#import "ComicModel.h"
#import <SDImageCache.h>

#define WIDTH [UIScreen mainScreen].bounds.size.width

#define HEIGHT [UIScreen mainScreen].bounds.size.height

@implementation ComicCell

- (instancetype)initWithStyle:(UITableViewCellStyle)style reuseIdentifier:(NSString *)reuseIdentifier {
    
    if (self = [super initWithStyle:style reuseIdentifier:reuseIdentifier]) {
        
        [self createCellUI];
        
    }
    
    return self;
}


+ (instancetype)cellWithTableView:(UITableView *)tableView {
    
    ComicCell *cell = [tableView dequeueReusableCellWithIdentifier:@"ID"];
    
    if (!cell) {
        
        
        cell = [[ComicCell alloc] initWithStyle:UITableViewCellStyleDefault reuseIdentifier:@"ID"];
        
    }
    
    return cell;
    
}

- (void)createCellUI {
    
    self.coverView = [[UIImageView alloc] initWithFrame:CGRectMake(10, 10, 105, 138)];
    
    self.coverView.layer.cornerRadius = 5;
    
    self.coverView.layer.masksToBounds = YES;
    
    self.titleLabel = [[UILabel alloc] initWithFrame:CGRectMake(CGRectGetMaxX(self.coverView.frame) + 10, 10, WIDTH - 135, 20)];
    
    self.titleLabel.font = [UIFont boldSystemFontOfSize:17];
    
    self.authouLabel = [[UILabel alloc] initWithFrame:CGRectMake(CGRectGetMaxX(self.coverView.frame) + 10, CGRectGetMaxY(self.titleLabel.frame) + 10, WIDTH - 135, 14)];
    
    self.authouLabel.font = [UIFont systemFontOfSize:16];
    
    self.descLabel = [[UILabel alloc] initWithFrame:CGRectMake(CGRectGetMaxX(self.coverView.frame) + 10, CGRectGetMaxY(self.authouLabel.frame) + 10, WIDTH - 135, 50)];
    
    self.descLabel.numberOfLines = 0;
    
    self.descLabel.font = [UIFont systemFontOfSize:12];
    
    self.updateLabel = [[UILabel alloc] initWithFrame:CGRectMake(CGRectGetMaxX(self.coverView.frame) + 10, CGRectGetMaxY(self.descLabel.frame) + 10, WIDTH - 135, 20)];
    
    self.updateLabel.font = [UIFont systemFontOfSize:13];
    
    [self.contentView addSubview:self.coverView];
    
    [self.contentView addSubview:self.titleLabel];
    
    [self.contentView addSubview:self.authouLabel];
    
    [self.contentView addSubview:self.descLabel];
    
    [self.contentView addSubview:self.updateLabel];
    
    
}


- (void)configCellWithModel:(ComicModel *)model {
    
    [self.coverView sd_setImageWithURL:[NSURL URLWithString:model.images] completed:^(UIImage *image, NSError *error, SDImageCacheType cacheType, NSURL *imageURL) {
       
        [[SDImageCache sharedImageCache] storeImage:image forKey:model.comicId];
    }];
    
    self.titleLabel.text = model.name;
    
    self.authouLabel.text = [NSString stringWithFormat:@"作者:%@", model.author];
    
    self.descLabel.text = [NSString stringWithFormat:@"简介:%@", model.introduction];
    
    self.updateLabel.textColor = [UIColor redColor];
    
    if ([model.updateType isEqualToString:@"1"]) {
        
        self.updateLabel.text = [NSString stringWithFormat:@"%@  每周%@更新", model.updateInfo, model.updateValue];
        
    }else if ([model.updateType isEqualToString:@"2"]) {
        
        self.updateLabel.text = [NSString stringWithFormat:@"%@  每月%@日更新", model.updateInfo, model.updateValue];
        
    }else if ([model.updateType isEqualToString:@"3"]) {
        
        self.updateLabel.text = [NSString stringWithFormat:@"%@  每周%@更新", model.updateInfo, model.updateValue];
        
    }else if ([model.updateType isEqualToString:@"4"]) {
        
        self.updateLabel.text = [NSString stringWithFormat:@"%@  更新%@", model.updateInfo, model.updateValue];
        
    }else if ([model.updateType isEqualToString:@"5"]) {
        
        self.updateLabel.text = [NSString stringWithFormat:@"%@  %@更新", model.updateInfo, model.updateValue];
        
    }else if ([model.updateType isEqualToString:@"6"]){
        
        self.updateLabel.text = model.updateInfo;
        
    }
    
    
    
}


- (void)configSearchCellWithModel:(ComicModel *)model {
    
    [self.coverView sd_setImageWithURL:[NSURL URLWithString:model.images]];
    
    [[SDImageCache sharedImageCache] storeImage:self.coverView.image forKey:model.comicId];
    
    self.titleLabel.text = model.name;
    
    self.authouLabel.text = [NSString stringWithFormat:@"作者:%@", model.author];
    
    self.descLabel.text = [NSString stringWithFormat:@"简介:%@", model.introduction];
    
    self.updateLabel.textColor = [UIColor redColor];
    
    if ([model.updateType isEqualToString:@"1"]) {
        
        self.updateLabel.text = [NSString stringWithFormat:@"每周%@更新", model.updateValue];
        
    }else if ([model.updateType isEqualToString:@"2"]) {
        
        self.updateLabel.text = [NSString stringWithFormat:@"每月%@日更新", model.updateValue];
        
    }else if ([model.updateType isEqualToString:@"3"]) {
        
        self.updateLabel.text = [NSString stringWithFormat:@"每周%@更新", model.updateValue];
        
    }else if ([model.updateType isEqualToString:@"4"]) {
        
        self.updateLabel.text = [NSString stringWithFormat:@"更新%@", model.updateValue];
        
    }else if ([model.updateType isEqualToString:@"5"]) {
        
        self.updateLabel.text = [NSString stringWithFormat:@"%@新", model.updateValue];
        
    }
    
    
    
}


@end

File: Personal/小项目/ComicAndPicture/ComicAndPicture/View/ComicCell.swift
//
//  ComicCell.swift
//  ComicAndPicture
//
//  Created by vcyber on 16/11/4.
//  Copyright © 2016年 vcyber. All rights reserved.
//

import UIKit

class ComicCell: UITableViewCell {

    override func awakeFromNib() {
        super.awakeFromNib()
        
    }
    
    private var coverView: UIImageView?
    private var titleLable: UILabel?
    private var authouLabal: UILabel?
    private var descLabel: UILabel?
    private var updateLabel: UILabel?
    
    
    override init(style: UITableViewCellStyle, reuseIdentifier: String?) {
        super.init(style: style, reuseIdentifier: reuseIdentifier)
        self.coverView = UIImageView(frame: CGRect(x: 10, y: 10, width: 105, height: 138))
        self.coverView?.layer.cornerRadius = 5
        self.coverView?.layer.masksToBounds = true
        
        self.titleLable = UILabel(frame: CGRect(x: (self.coverView?.frame.maxX)! + 10, y: 10, width: SCREEN_WIDTH - 135, height: 20))
        titleLable?.font = UIFont.systemFont(ofSize: 17)
        
        self.authouLabal = UILabel(frame: CGRect(x: (self.titleLable?.frame.origin.x)!, y: (self.titleLable?.frame.maxY)! + 10, width: SCREEN_WIDTH - 135, height: 14))
        self.authouLabal?.font = UIFont.systemFont(ofSize: 16)
        
        self.descLabel = UILabel(frame: CGRect(x: (self.titleLable?.frame.origin.x)!, y: (self.authouLabal?.frame.maxY)! + 10, width: SCREEN_WIDTH - 135, height: 50))
        self.descLabel?.numberOfLines = 0
        self.descLabel?.font = UIFont.systemFont(ofSize: 12)
        
        self.updateLabel = UILabel(frame: CGRect(x: (self.titleLable?.frame.origin.x)!, y: (self.descLabel?.frame.maxY)! + 10, width: SCREEN_WIDTH - 135, height: 20))
        self.updateLabel?.font = UIFont.systemFont(ofSize: 13)
        self.updateLabel?.textColor = UIColor.red
        
        self.contentView.addSubview(self.coverView!)
        self.contentView.addSubview(self.titleLable!)
        self.contentView.addSubview(self.authouLabal!)
        self.contentView.addSubview(self.descLabel!)
        self.contentView.addSubview(self.updateLabel!)
        
    }
    
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    static func createCell(tableView:UITableView, reuseIdentifier: String) -> ComicCell {
        var cell:ComicCell? = tableView.dequeueReusableCell(withIdentifier: reuseIdentifier) as? ComicCell
        if cell == nil {
            cell = ComicCell(style: UITableViewCellStyle.default, reuseIdentifier: reuseIdentifier)
        }
        return cell!
    }

    func configCell(model:ComicModel) -> Void {
        if SDImageCache.shared().diskImageExists(withKey: model.comicId) {
            coverView?.image = SDImageCache.shared().imageFromDiskCache(forKey: model.comicId)
        }else {
            coverView?.sd_setImage(with: URL(string: model.images) as URL!, completed: { (image, error, cacheType, url) in
                SDImageCache.shared().store(image, forKey: model.comicId, toDisk: true)
            })
        }
        self.titleLable?.text = model.name
        self.authouLabal?.text = "作者：" + model.author
        self.descLabel?.text = "简介：" + model.introduction
        switch model.updateType {
        case "1", "3":
            self.updateLabel?.text = String(format: "%@  每周%@更新", model.updateInfo, model.updateValue)
        case "2":
            self.updateLabel?.text = String(format: "%@  每月%@日更新", model.updateInfo, model.updateValue)
        case "4":
            self.updateLabel?.text = String(format: "%@  更新%@", model.updateInfo, model.updateValue)
        case "5":
            self.updateLabel?.text = String(format: "%@  %@更新", model.updateInfo, model.updateValue)
        default:
            self.updateLabel?.text = model.updateInfo
        }
    }
    
    override func setSelected(_ selected: Bool, animated: Bool) {
        super.setSelected(selected, animated: animated)

        // Configure the view for the selected state
    }

}
